File: 6159.py
input=open(0).readline
n,s=map(int,input().split())
buf=[]
for _ in range(n): buf.append(int(input()))
count=0
for i in range(n):
    for j in range(i+1,n):
        a,b=buf[i],buf[j]
        if a+b<=s: count+=1
print(count)

File: 6159.c
#include <stdio.h>
int main(){
    int n,s;
    scanf("%d %d",&n,&s);
    int buf[20000];
    for (int i=0; i<n; i++) scanf("%d",&buf[i]);
    int count=0;
    int a,b;
    for (int i=0; i<n; i++){
        for (int j=i+1; j<n; j++){
            a=buf[i];
            b=buf[j];
            if (a+b<=s) count+=1;
        }
    }
    printf("%d\n",count);
    return 0;
}